chain/default_syncer.go

```diff
@@ -312,31 +315,24 @@ func (syncer *DefaultSyncer) widen(ctx context.Context, ts types.TipSet) (types.
 	return wts, nil
 }
 
-// HandleNewBlocks extends the Syncer's chain store by the given blocks if they
+// HandleNewTipset extends the Syncer's chain store with the given tipset if they
 // represent a valid extension. It limits the length of new chains it will
 // attempt to validate and caches invalid blocks it has encountered to
 // help prevent DOS.
-func (syncer *DefaultSyncer) HandleNewBlocks(ctx context.Context, blkCids []cid.Cid) error {
-	// ********** WARNING **********
-	//
-	// This concurrency model is flawed.  The mutex is held during a possibly
-	// long call to the network.  TODO: re-evaluate / re-design the concurrency
-	// model to allow for collectChain to be called outside the lock.
-	// FYI the two biggest concurrency concerns at present would be addressed
-	// by locking after collectChain completes, so my hunch is this can be
-	// fixed by simply moving the lock call below collectChain.
-	logSyncer.Debugf("trying to sync %v\n", blkCids)
+func (syncer *DefaultSyncer) HandleNewTipset(ctx context.Context, tipsetCids types.SortedCidSet) error {
+	logSyncer.Debugf("trying to sync %v\n", tipsetCids)
 	syncer.mu.Lock()
```

Comment:
I would definitely add a comment here talking about the tradeoffs of moving this lock, otherwise you will just be having this conversation with someone over again when they try to change it in the future.